src/main.rs

```diff
@@ -74,26 +76,39 @@ fn main() -> Result<(), GitError> {
         spinner.set_message("deleting branches...");
         spinner.enable_steady_tick(160);
 
-        let mut deletion_errors: Vec<String> = vec![];
-        for branch_name in gone_branches {
-            spinner.set_message(&format!("deleting \"{}\"...", branch_name));
-            match git.delete(args.force, branch_name) {
-                Err(GitError::CommandError(msg)) => deletion_errors.push(msg),
-                Err(GitError::ExecError(msg)) => deletion_errors.push(msg),
-                Err(GitError::ParseError(msg)) => deletion_errors.push(msg),
-                Err(GitError::UnknownError) => {
-                    deletion_errors.push(String::from("Unknown error encountered."))
-                }
-                _ => (),
+        let (deleted_branches, deletion_errors) =
+            stale_branches
+                .into_iter()
+                .fold((vec![], vec![]), |(mut del, mut err), branch_name| {
+                    spinner.set_message(&format!("deleting \"{}\"...", branch_name));
+                    match git.delete(args.force, branch_name) {
+                        Err(GitError::CommandError(msg)) => err.push((branch_name, msg)),
+                        Err(GitError::ExecError(msg)) => err.push((branch_name, msg)),
+                        Err(GitError::ParseError(msg)) => err.push((branch_name, msg)),
+                        Err(GitError::UnknownError) => err.push((
+                            branch_name,
+                            String::from("An unknown error was encountered"),
+                        )),
+                        _ => del.push(branch_name),
```

Comment:
```
use of moved value: `branch_name`
value used here after moverustcE0382
main.rs(84, 50): value moved here
main.rs(82, 62): move occurs because `branch_name` has type `String`, which does not implement the `Copy` trait
```